### PR TITLE
Robotic Analyzer Buff + Cleanup

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

This PR does a few things regarding the robotic analyzer
1. Moves the code for robotic analyzers from components.dm, to scanners.dm. 
2. It replaces the many to_chat calls with list operations. 
3. It adds the function to scan AI cores to see their health.
4. It will now show temperature and reagents of scanned IPCs.
5. It improves readability for all the modes, including only showing damaged limbs for IPCs.
## Why It's Good For The Game

1. It keeps files organized, it didn't make sense for it to be with the code for components.
2. Currently, analyzing an IPC noticeably lags the game because of how many to_chat procs it calls. Using lists makes sending large chat messages much faster.
3. There's no good way to check an AI's exact health without asking the AI, even as a ghost.
4. Health analyzers show temperature, but currently there's no way to check the reagents of an IPC. This change both makes sense for the item to do IC, and fills both of those roles.
5. The boxes look good, and filling chat with messages that all of your limbs are fine is uneeded. This trims the information down to just the important stuff. Humans will always have their prosthetics show up on the scanner even if they are undamaged, however.
## Images of changes

image
## Testing

Beat up a bunch of non-organics, then scanned them. Made sure ghost analyzing worked fine, made sure scanning humans gets their prosthetics.

## Changelog
:cl:
tweak: Robotic analyzer won't lag, looks better, and has more info now.
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -328,6 +329,131 @@ REAGENT SCANNER
 	origin_tech = "magnets=2;biotech=2"
 	usesound = 'sound/items/deconstruct.ogg'
 
+/obj/item/robotanalyzer
+	name = "cyborg analyzer"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "robotanalyzer"
+	item_state = "analyzer"
+	desc = "A hand-held scanner able to diagnose robotic injuries."
```
Comment (Henri215) [inline code/game/objects/items/devices/scanners.dm:337]:
```suggestion
/obj/item/robotanalyzer
	name = "cyborg analyzer"
	desc = "A hand-held scanner able to diagnose robotic injuries."
	icon = 'icons/obj/device.dmi'
	icon_state = "robotanalyzer"
	item_state = "analyzer"
```

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -328,6 +329,131 @@ REAGENT SCANNER
...
+		var/list/msgs = list()
+		user.visible_message("<span class='warning'>[user] has analyzed the floor's components!</span>", "<span class='warning'>You try to analyze the floor's vitals!</span>")
+		msgs += "<span class='info'>Analyzing Results for The floor:\n\t Overall Status: Unknown</span>"
+		msgs += "<span class='info'>\t Damage Specifics: <font color='#FFA500'>[0]</font>/<font color='red>[0]</font></span>"
+		msgs += "<span class='info'>Key: <font color='#FFA500'>Burns</font><font color ='red'>/Brute</font></span>"
+		msgs += "<span class='info'>Chassis Temperature: ???</span>"
+		to_chat(user, chat_box_healthscan(msgs.Join("<br>")))
+		return
+
+	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components.</span>","<span class='notice'>You have analyzed [M]'s components.</span>")
```
Comment (Henri215) [inline code/game/objects/items/devices/scanners.dm:357]:
```suggestion
	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components.</span>", "<span class='notice'>You have analyzed [M]'s components.</span>")
```

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -328,6 +329,131 @@ REAGENT SCANNER
...
+		msgs += "<span class='info'>Chassis Temperature: ???</span>"
+		to_chat(user, chat_box_healthscan(msgs.Join("<br>")))
+		return
+
+	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components.</span>", "<span class='notice'>You have analyzed [M]'s components.</span>")
+	robot_healthscan(user, M)
+	add_fingerprint(user)
+
+
+/proc/robot_healthscan(mob/user, mob/living/M)
```
Comment (Sirryan2002) [inline code/game/objects/items/devices/scanners.dm:362]: what an actual abomination of a proc

**Review by Sirryan2002 — APPROVED** [review]

File: code/game/objects/items/devices/scanners.dm [hunk 4]
```diff
@@ -328,6 +329,131 @@ REAGENT SCANNER
...
+	item_state = "analyzer"
+	flags = CONDUCT
+	slot_flags = SLOT_FLAG_BELT
+	throwforce = 3
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 5
+	throw_range = 10
+	origin_tech = "magnets=1;biotech=1"
+
+/obj/item/robotanalyzer/attack(mob/living/M as mob, mob/living/user as mob)
```
Comment (lewcc) [inline code/game/objects/items/devices/scanners.dm:346]:
```suggestion
/obj/item/robotanalyzer/attack(mob/living/M, mob/living/user)
```

**Review by lewcc — COMMENTED** [review]
Yeah, this is pretty nasty, but I don't think you need to be the one to fix some of the spacing -- unless you want to, of course.

**Pull request merged**